test/org/javarosa/core/model/Safe2014DagImplTest.java

```diff
@@ -0,0 +1,104 @@
+package org.javarosa.core.model;
+
+import org.javarosa.core.model.instance.FormInstance;
+import org.javarosa.core.model.instance.TreeElement;
+import org.javarosa.core.model.instance.TreeReference;
+import org.javarosa.debug.Event;
+import org.javarosa.debug.EventNotifier;
+import org.junit.Test;
+
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.hamcrest.CoreMatchers.instanceOf;
+import static org.hamcrest.core.IsEqual.equalTo;
+import static org.javarosa.test.utils.FormParserHelper.parse;
+import static org.javarosa.test.utils.ResourcePathHelper.r;
+import static org.junit.Assert.assertThat;
+
+public class Safe2014DagImplTest {
+
+    private final List<Event> dagEvents = new ArrayList<>();
+
+    private final EventNotifier eventNotifier = new EventNotifier() {
+
+        @Override
+        public void publishEvent(Event event) {
+            dagEvents.add(event);
+        }
+    };
+
+    @Test
+    public void deleteRepeatGroup_evaluatesTriggerables_dependedOnAllRepeatGroupSiblings() throws Exception {
+        // Given
+        final FormDef formDef =
+                parse(r("org/javarosa/core/model/repeat-group-with-position-calculation.xml")).formDef;
+
+        formDef.setEventNotifier(eventNotifier);
+        IDag dag = getIDagImplUnderTest(formDef);
+        assertThat(dag, instanceOf(Safe2014DagImpl.class));
```

Comment:
getIDagImplUnderTest casts to this, so you’ll crash before failing this assertion.

Reply:
@dcbriccetti Nice catch! Thanks

Reply:
@dcbriccetti Fixed